syntax = "proto3";

package biotracker;

message Feature {
  repeated SkeletonNode nodes = 2;
  float score = 3;
}

message Features {
  repeated Feature features = 2;
  SkeletonDescriptor skeleton = 3;
}

message Entity {
  string id = 1;
  uint32 frame_number = 2;
  Feature feature = 3;
}

message Entities {
  repeated Entity entities = 2;
  SkeletonDescriptor skeleton = 3;
}

message Track {
  SkeletonDescriptor skeleton = 3;
  repeated Entity observations = 1;
}

message SkeletonNode {
  float x = 1;
  float y = 2;
  float score = 3;
}

message SkeletonEdge {
  uint32 source = 1;
  uint32 target = 2;
}

message SkeletonDescriptor {
  repeated SkeletonEdge edges = 1;
  repeated string node_names = 2;
  uint32 center_node_index = 3;
  uint32 front_node_index = 4;
}
